### ISSUE TITLE
Suppressing output on jupyter notebooks for fit

### ISSUE THREAD

**Issue by AtreyeeS** [gammapy/gammapy#1542]
Is there any way to suppress the output on the screen for 

`fit = SkyModelMapFit(
    model=model.copy(),
    counts=counts_map,
    exposure=exposure_map,
)`

`fit.fit()`

This prints results on the screen, and can be very disturbing when fitting inside a loop (eg: for a large number of simulations)

**Comment by cdeil** [comment]
I think for this one has to pass `print_level=0` to `Minuit`.
@joleroi - Is it possible to pass such options at the moment?

**Comment by cdeil** [comment]
@dcfidalgo - Just one more thought on this how to configure the Minuit fit.

One option is to add option dicts that are passed along.

But another option might be to change this:

https://github.com/gammapy/gammapy/blob/1a43d653ac8a8a528c7dbd7fc4c61160a3e1e773/gammapy/utils/fitting/iminuit.py#L42-L46

It's probably not a good idea to create `Minuit` and call `migrad` in the same function.
Note that `migrad` also has options the user might want to control:
https://iminuit.readthedocs.io/en/latest/api.html#iminuit.Minuit.migrad

Is it possible to adapt the Gammapy call, so that on SpectrumFit or MapFit class init, only the `Minuit` init happens (that's very cheap)?
And then users have an easier time to configure / control the execution, e.g. like this:
```
fit = SpectrumFit()
fit.minuit.set_printlevel(0)
fit.minuit.configure_what_you_like()
fit.run() # calls migrad
fit.minuit.do_what_you_like()
```

I realise that this is not the long term solution, it hardcodes the minuit interface. But it gives us power for now, and then we can build a proper generic fitting backend (probably with a thin layer of wrapper classes for `Optimiser` and `ErrorEstimator` and `ParameterScaler` or whatever we like) in the coming months.

@dcfidalgo - Up to you how to implement this. I just wanted to mention the option of splitting out the `migrad` call from the function where the `Minuit` init happens.

**Comment by dcfidalgo** [comment]
@cdeil Sorry, i just read your response now. I already submitted a PR where i pass along a `dict` ...
Let's go with the `dict` for now and leave major changes to some long term solution.

**Comment by cdeil** [comment]
Resolved by @dcfidalgo in #1553.
